# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/loader/ModuleLoaderBase.cpp
# Commit: c6c863899bf9
# Full Hash: c6c863899bf944a2cb49531cd7a9a7c752b6f700
# Author: Yoshi Cheng-Hao Huang <allstars.chh@gmail.com>
# Date: 2025-09-26 06:07 +0000
# Regressor Bug: 1968890
# File Overlap Count: 3
# Overlapping Files: js/loader/ModuleLoadRequest.h, js/loader/ModuleLoadRequest.cpp, js/loader/ModuleLoaderBase.cpp
# Description:
#   Bug 1968890 - Part 5: Add ModuleLoadRequest::SetErroredLoadingImports(). r=jonco
# ==============================================================================

diff -r 52a87b60e974 -r c6c863899bf9 js/loader/ModuleLoaderBase.cpp
--- a/js/loader/ModuleLoaderBase.cpp	Fri Sep 26 06:07:44 2025 +0000
+++ b/js/loader/ModuleLoaderBase.cpp	Fri Sep 26 06:07:44 2025 +0000
@@ -1325,6 +1325,7 @@
                                 JSMSG_DYNAMIC_IMPORT_FAILED, url.get());
       FinishLoadingImportedModuleFailedWithPendingException(aCx, payload);
     }
+    aRequest->SetErroredLoadingImports();
   } else if (moduleScript && !error.isUndefined()) {
     LOG(
         ("ScriptLoadRequest (%p): LoadRequestedModules rejected: set error to "
